expression/value.go

```diff
@@ -45,14 +47,19 @@ func (l Value) IsStatic() bool {
 
 // String implements the Expression String interface.
 func (l Value) String() string {
-	switch x := l.Val.(type) {
-	case nil:
+	if types.IsNil(l.Val) {
 		return "NULL"
+	}
+	switch x := l.Val.(type) {
 	case string:
 		return fmt.Sprintf("%q", x)
-	default:
-		return fmt.Sprintf("%v", l.Val)
+	case *types.DataItem:
+		if x.Type.Tp == mysql.TypeString {
+			return fmt.Sprintf("%q", x.Data)
+		}
+		return fmt.Sprintf("%v", x.Data)
 	}
+	return fmt.Sprintf("%v", l.Val)
```

Comment:
For string literal, we need to print a string with quotation mark, so we need use %q.